engine/src/main/java/org/terasology/rendering/dag/stateChanges/BindFBO.java

```diff
@@ -38,6 +54,13 @@ public StateChange getDefaultInstance() {
 
     @Override
     public RenderPipelineTask generateTask() {
-        return null;
+        task = new BindFBOTask(fboId);
+        return task;
+    }
+
+    @Override
+    public void update() {
+        fboId = frameBuffersManager.getFBO(this.getValue()).fboId;
+        task.setFboToBind(fboId);
```

Comment:
I imagine that -if- the state change is redundant it will remain in the intermediate pipeline list but won't generate a task and the variable task will contain null, generating a NPE.

However, rather than checking if task is null, how about we subscribe to the FBM -only- if a task is generated?

Also, I now realize that a state change in the intermediate list could be redundant one moment, needed another and back to redundant as the user changes settings. In particular, if a needed state change is made redundant, there is no functionality to unsubscribe from the FBM until the state becomes needed again. I think it's ok to leave the task in memory

Finally, how about StateChange.value becomes "protected" so that you can write:

`fboId = frameBuffersManager.getFBO(value).fboId;`

That been said, I really wish we didn't go through an AbstractStateChange so that we could write:

`fboId = frameBuffersManager.getFBO(fboName).fboId;`